public/app/features/provisioning/api/endpoints.gen.ts

```diff
@@ -759,20 +759,32 @@ export type S3RepositoryConfig = {
   bucket?: string;
   region?: string;
 };
+export type SyncOptions = {
+  /** Enabled must be saved as true before any sync job will run */
+  enabled: boolean;
+  /** When non-zero, the sync will run periodically */
+  intervalSeconds?: number;
+  /** Where values should be saved
+    
+    Possible enum values:
+     - `"instance"` Resources are saved in the global context Only one repository may specify the `instance` target When this exists, the UI will promote writing to the instance repo rather than the grafana database (where possible)
+     - `"mirror"` Resources will be saved into a folder managed by this repository The folder k8s name will be the same as the repository k8s name It will contain a copy of everything from the remote */
+  target: 'instance' | 'mirror';
+};
 export type RepositorySpec = {
-  /** Describe the feature toggle */
+  /** Repository description */
   description?: string;
   /** Edit options within the repository */
   editing: EditingOptions;
-  /** The folder that is backed by the repository. The value is a reference to the Kubernetes metadata name of the folder in the same namespace. */
-  folder?: string;
```

Comment:
Where do we get the folder data from? There's quite a bit of code depending on this property.

Reply:
when the target is `mirror` the folder will have the same id as the repository.

when it is 'instance' it is everything -- so not really relevant.

I updated most of the code to search for folders matching the repo.metadata.name -- with https://github.com/grafana/grafana/pull/99791 this will be easier to tell the difference between global vs repository